File: New Attempts/Fortran/Problem_4/Problem_4/Problem_4.f90
program Problem_4
    implicit none
    
    integer :: number, temp, i, j
    
    number = 0
    temp = 0
    
    do i = 1000, 100, -1
        do j = 1000, 100, -1
            temp = i * j
            if (temp .gt. number) then
                if (palindrome(temp)) then
                    number = temp
                end if
            end if
        end do
    end do
    
    print *,number
    
    read(*,*)
    
    contains
        logical function palindrome(n)
            integer, intent(in) :: n
            integer :: x, y
    
            x = n
            y = 0
    
            do while (x .gt. 0)
                y = y * 10
                y = y + mod(x, 10)
                x = x / 10
            end do
    
            if(y .eq. n) then
                palindrome = .true.
            else
                palindrome = .false.
            end if
    
            return
        end  
end program Problem_4

File: New Attempts/Python/Problem_4/Problem_4/Problem_4.py
def Palindrome(number):
    y, temp = 0, number
    while temp > 0:
        y += temp % 10
        temp /= 10

    if y == number:
        return True
    return False

number = 0

for i in range(1000, 100, -1):
    for j in range(1000, 100, -1):
        if (i * j) > number and Palindrome(i * j):
            number = (i * j)

print (number)